syntax = "proto3";

package profile;

option go_package = "github.com/SzymonSt/autoinstrumentation-playground/dummy-proto/profileservice/v1";
option java_package = "com.szymonst.profileservice.v1";
option java_outer_classname = "ProfileServiceProto";

service ProfileService {
    rpc GetProfile(GetProfileRequest) returns (GetProfileResponse);
    rpc SetProfile(SetProfileRequest) returns (SetProfileResponse);
}

message Profile{
    string id = 1;
    string name = 2;
    string email = 3;
}

message GetProfileRequest {
  string id = 1;
}

message GetProfileResponse {
    Profile profile = 1;
}

message SetProfileRequest {
    string name = 1;
    string email = 2;
}

message SetProfileResponse {
    Profile profile = 1;
}